Block Hooks: Revert opening up hooked blocks to all block themes

nerrad opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

In #45581, the WooCommerce implementation of the WP Block Hooks API auto-injects the mini-cart and customer account block on all sites with a block theme (running WooCommerce of course). After the release of WC 8.8 (where this change was introduced), we've discovered that impact on existing sites is greater than expected and are reverting until we can implement this more safely.

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

The only themes this should apply to now are: "Twenty Twenty-Four, Twenty Twenty-Three, Twenty Twenty-Two, Tsubaki, Zaino, Thriving Artist, Amulet, Tazza" which were where the hooks were implemented as of WC 8.7.

1. Make sure that the mini-cart block is auto-injected in the header template parts and header patterns for the above list of themes.
2. Make sure that the mini-cart block is _not_ auto-injected in the header template parts and header patterns for any block theme not in the above list.

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

<!-- Choose only one -->

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Reverts auto-injecting specific Woo Blocks in every block theme and restores only auto-injecting in themes found in the allow list.

#### Comment <!-- If the changes in this pull request don't warrant a changelog entry, you can alternatively supply a comment here. Note that comments are only accepted with a significance of "Patch" -->

</details>

nerrad committed:
Revert opening up hooked blocks to all block themes.

Commit:
Add changefile(s) from automation for the following project(s): woocommerce

danielwrobert reviewed (approved):
LGTM!

The code looks good.

I've tested against two themes outside of the `$theme_include_list` (Portia and Alara) and the hooked blocks are no longer inserted with this PR applied.

I've also tested against all of the themes in the `$theme_include_list` and the hooked blocks _are_ inserted, as expected.

Based on this, I am approving the PR.

That being said, I noticed that in Amulet and Tazza, the positioning of the inserted blocks is a bit wonky (see screenshots below). This appears to mainly concern the My Account block, as opposed to the Mini-Cart so it seems best to create an issue and address that separately. But I wanted to note here, as I discovered it in testing for this PR.

**Amulet:**
![CleanShot 2024-04-25 at 15 50 02](https://github.com/woocommerce/woocommerce/assets/481776/2b848b15-00a6-4ae4-8339-b1fc58c00890)

**Tazza:**
![CleanShot 2024-04-25 at 15 50 41](https://github.com/woocommerce/woocommerce/assets/481776/b390a26e-3163-4d3b-888f-2b85f8e797a5)

danielwrobert committed:
Add version to @since tag in fn comment.

Missing @since version throwing a CI check error:

`A "@since" tag was found but no version declared.`

Adding `8.4.0`, as this was the version referenced before the revert.
